src/navigators/Main.js

```diff
@@ -41,7 +40,6 @@ const MainNavigator = () => {
 
   useEffect(() => {
     const timer = setTimeout(async () => {
```

Comment:
You could remove this from `setTimeout` and see if the status bar color is being set during the launch itself.